### ISSUE TITLE
Prisma client not working with aws lambda

### ISSUE THREAD

**Issue by amitleuva1987** [serverless-nextjs/serverless-next.js#2459]
### Issue Summary

**I am not sure if it is bug or a new feature request.**
I am using next js "12.1.6". I have installed "@prisma/client": "^3.15.2" . I am trying to go serverless with aws. when I start deploying the app, it doesn't show any error. It creates "lambda functions" and static stuff in "aws s3" and provides a cloudfront url. when I visit the deployed app, it show me 503 error. If I remove the prisma portion, the app starts running properly.

FYI : I use mysql database created on "railway.app.". I hope serverless support railway.app databases when deployed with aws lambda.   

### Actual behavior

the deployed app return 503 error. I am new to aws and not able to see the logs.
### Expected behavior

It should return a list of users in index page. below is the code to display users from table.
```
export async function getServerSideProps(content) {
  const prisma = new PrismaClient()
  const users = await prisma.user.findMany({
    select: {
      id: true,
      email: true,
    }
  })

  return {
    props : { users }
  }
}
```
### Steps to reproduce

I already have the repo over github which is running fine with vercels. below is the git repo url.
https://github.com/amitleuva1987/full-stack-nextjs-with-prisma
### Screenshots/Code/Configuration/Logs
below is my serverless.yml file

```
myNextApp:
  component: "@sls-next/serverless-component"
```
below is .env file
```
NEXTAUTH_URL="https://d1vl8g8lv7yuwi.cloudfront.net"  // I have taken the app down, this url may not work


DATABASE_URL="********"  // prisma proxy url

MIGRATE_DATABASE_URL="*****"  // railay.app mysql db

PRISMA_CLIENT_ENGINE_TYPE='dataproxy'
GIT_ID="*********"
GIT_SECRET="********"

EMAIL_SERVER="******"
EMAIL_FROM=NextAuth <noreply@example.com>
```

### Versions

    "@next-auth/prisma-adapter": "^1.0.3",
    "@prisma/client": "^3.15.2",
    "bcryptjs": "^2.4.3",
    "bootstrap": "^5.1.3",
    "next": "12.1.6",
    "next-auth": "4.8.0",
    "nodemailer": "^6.7.6",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "swr": "^1.3.0"

### Additional context

The app works very well with vercels and in my local pc. Below is an open issue which needs attention. 
[issue url](https://github.com/prisma/prisma/issues/12702)

Below repo is one of the promising solution which did not work in my case.
[repo url](https://github.com/millsp/prisma-serverless-nextjs)

### Checklist

- [x] I have reviewed the [README](https://github.com/serverless-nextjs/serverless-next.js/blob/master/README.md) and [FAQs](https://github.com/serverless-nextjs/serverless-next.js#faq), which answers several common questions.
- [x] I have reviewed our [DEBUGGING](https://github.com/serverless-nextjs/serverless-next.js/wiki/Debugging-Issues) wiki and have tried my best to include complete information and reproduction steps (including your configuration) as is possible. As there is only one maintainer (who maintains this in his free time) and thus very limited resources, if you have time, please try to debug the issue a bit yourself if possible.
- [x] You have first tried using the most recent `latest` or `alpha` `@sls-next/serverless-component` [release version](https://github.com/serverless-nextjs/serverless-next.js/releases), which may have already fixed your issue or implemented the feature you are trying to use. Note that the old `serverless-next.js` component and the `serverless-next.js` plugin are deprecated and no longer maintained.

**Comment by shengslogar** [comment]
Same issue on Prisma v4. Here's our log trace:

```
{
    "errorType": "Error",
    "errorMessage": "ENOENT: no such file or directory, open '/var/task/chunks/schema.prisma'",
    "code": "ENOENT",
    "errno": -2,
    "syscall": "open",
    "path": "/var/task/chunks/schema.prisma",
    "clientVersion": "4.0.0",
    "stack": [
        "Error: ENOENT: no such file or directory, open '/var/task/chunks/schema.prisma'",
        "    at Object.openSync (node:fs:585:3)",
        "    at Object.readFileSync (node:fs:453:35)",
        "    at new LibraryEngine (/var/task/chunks/1530.js:45592:41)",
        "    at PrismaClient.getEngine (/var/task/chunks/1530.js:50537:16)",
        "    at new PrismaClient (/var/task/chunks/1530.js:50509:29)",
        "    at Object.93163 (/var/task/pages/api/login.js:325:19)",
        "    at __webpack_require__ (/var/task/webpack-api-runtime.js:25:43)",
        "    at Module.34661 (/var/task/pages/api/login.js:212:71)",
        "    at __webpack_require__ (/var/task/webpack-api-runtime.js:25:43)",
        "    at Module.70601 (/var/task/pages/api/login.js:378:23)"
    ]
}
```

Because Lambdas deployed in this context are regionless, you'll have to look for CloudWatch logs in the AWS region closest in physical proximity to the device making the HTTP request. Tripped me up, too.

**Comment by amitleuva1987** [comment]
@shengslogar , Thank you for your efforts to solve this issue. I would try it today and will revert you with the outcome.

**Comment by amitleuva1987** [comment]
@shengslogar , I just checked my error logs in Cloudwatch and it looks very simillar to your error logs.

ERROR	Unhandled error during request: Error: ENOENT: no such file or directory, open '/var/task/chunks/schema.prisma'
    at Object.openSync (fs.js:498:3)
    at Object.readFileSync (fs.js:394:35)
    at new LibraryEngine (/var/task/chunks/281.js:44798:41)
    at PrismaClient.getEngine (/var/task/chunks/281.js:49711:16)
    at new PrismaClient (/var/task/chunks/281.js:49683:29)
    at getServerSideProps (/var/task/pages/index.js:308:20)
    at Object.renderToHTML (/var/task/chunks/379.js:9545:26)
    at async renderReqToHTML (/var/task/chunks/379.js:2331:26)
    at async Module.render (/var/task/chunks/379.js:2460:30)
    at async renderRoute (/var/task/default-handler-0a0dc009.js:1866:13) {
  errno: -2,
  syscall: 'open',
  path: '/var/task/chunks/schema.prisma',
  clientVersion: '3.15.2'
}

**Comment by amitleuva1987** [comment]
> useServerlessTraceTarget: true

should I put in under "inputs:"   in "serverless.yml"?

**Comment by shengslogar** [comment]
Correct! Here's an extract of ours:

```
slsApp:
  component: '@sls-next/serverless-component@3.7.0'

  inputs
    memory:
      defaultLambda: 512
      apiLambda: 512
      imageLambda: 512

    runtime:
      defaultLambda: 'nodejs16.x'
      apiLambda: 'nodejs16.x'
      imageLambda: 'nodejs16.x'

    domainMinimumProtocolVersion: 'TLSv1.2_2021'

    # Tell serverless-next.js to use Output File Tracing that `next build` would
    # normally do. Without this, our `schema.prisma` file won't be copied
    # and will error out in production.
    # @see https://nextjs.org/docs/advanced-features/output-file-tracing
    # @see https://github.com/serverless-nextjs/serverless-next.js#inputs
    # @see https://github.com/serverless-nextjs/serverless-next.js/issues/2459#issuecomment-1182938709
    useServerlessTraceTarget: true
```

**Comment by dedekpo** [comment]
After 2 days trying to figure it out I manage to fix it thanks to you. Thank you very much!!!

> Was able to resolve by adding `useServerlessTraceTarget: true` to my `serverless.yml`.

**Comment by amitleuva1987** [comment]
I am still receiving the same error even after using "useServerlessTraceTarget: true".  

will keep you guys updated. 

@shengslogar , can we run a "next js" app locally with "serverless-offline"?

**Comment by shengslogar** [comment]
@amitleuva1987  Wondering if you'd have any luck with Prisma v4? Never used `serverless-offline`, unfortunately.

**Comment by amitleuva1987** [comment]
@shengslogar , let me try Prisma v4.

**Comment by amitleuva1987** [comment]
@shengslogar , when I run serverless for the first time with debug mode, I see, it generated ".prisma/client" folder in "serverless/api-lambda/" but, I still receive below error.

file or directory, open '/var/task/chunks/schema.prisma'
    at Object.openSync (fs.js:498:3)
    at Object.readFileSync (fs.js:394:35)
    at new LibraryEngine (/var/task/chunks/281.js:45422:41)
    at PrismaClient.getEngine (/var/task/chunks/281.js:50367:16)
    at new PrismaClient (/var/task/chunks/281.js:50339:29)
    at getServerSideProps (/var/task/pages/index.js:307:20)
    at Object.renderToHTML (/var/task/chunks/379.js:9545:26)
    at async renderReqToHTML (/var/task/chunks/379.js:2331:26)
    at async Module.render (/var/task/chunks/379.js:2460:30)
    at async renderRoute (/var/task/default-handler-0a0dc009.js:1866:13) {
  errno: -2,
  syscall: 'open',
  path: '/var/task/chunks/schema.prisma',
  clientVersion: '4.0.0'
}

may be some caching issues, I will debug it will let you know regarding outcome.

**Comment by amitleuva1987** [comment]
@shengslogar , @dedekpo ,  guys, do you have any sample github repository for this (prisma with next)? 

also which databases are you using?

**Comment by shengslogar** [comment]
See if this helps you.

https://github.com/prisma/prisma-examples/tree/latest/javascript/rest-nextjs
https://github.com/prisma/prisma-examples/tree/latest/deployment-platforms/aws-lambda

MySQL 8 on my end. If you're still receiving the same error you posted earlier, then it's a config issue and not a database one.

**Comment by amitleuva1987** [comment]
@shengslogar , Thank you for your efforts. I came across few issues, similar to this one and I got to know that prisma still not working with next 12. have a look at this [issues](https://github.com/prisma/prisma/issues/10200). This issue has been closed but if you read last 3 replies, you would know that its not working with next 12. 

I tried your serverless.yml. I copied your serverless.yml but it's still not working. 

I am surprised that it works with your next 12.1.4. 

Anyways, I will put put some more efforts and let you know if I come across any promising solution
